add server side sampling docs (#5315)
getsentry/sentry-docs: src/docs/product/data-management-settings/server-side-sampling/getting-started.mdx

---
title: Getting Started
sidebar_order: 10
description: "Learn how to get started with server-side sampling."
---

<Note>

> This feature is in beta and is only available if your organization is participating in its limited release. Features in beta are still in-progress and may have bugs. We recognize the irony. If you’re interested in joining this limited release, please reach out at [abc@sentry.io](mailto:abc@sentry.io).

Author: ```suggestion
This feature is in beta and is only available if your organization is participating in its limited release. Features in beta are still in-progress and may have bugs. We recognize the irony. If you’re interested in participating, [join the waitlist](https://sentry.io/for/performance/#updates-signup).
```

- [Changing trace attributes in secondary transactions](/product/data-management-settings/server-side-sampling/current-limitations/#changing-trace-attributes-in-secondary-transactions)
- [Limited number of rule conditions](/product/data-management-settings/server-side-sampling/current-limitations/#limited-number-of-rule-conditions)
- [Arbitrary logical composition for rule conditions](/product/data-management-settings/server-side-sampling/current-limitations/#arbitrary-logical-composition-for-rule-conditions)

For more detailed information about each of these, check out our [full Current Limitations documentation](/product/data-management-settings/server-side-sampling/current-limitations/).

## 1. Update Sentry SDK Versions

Once you’ve confirmed that your application only relies on [supported SDKs](#current-limitations), then the next step is to update them to the specified version or higher. Check out the [migration documentation](#sampling-migration-guide) for the SDKs you’re using to learn everything you need to know to get up and running again with the latest Sentry features.

## 2. Set a Uniform Sampling Rate

> <Note>
> 
> If you’ve previously had sampling set in the SDK and you’re moving to server-side sampling, check out our migration guide.
> 
> </Note>
> 

Author: ```suggestion
```

You can use the following conditions to sample transactions by:

- Environment
- Release\*
- User Segment

\* Allows glob pattern matching

![Editing a sampling rule based on a release condition](edit-rule-modal.png)

If we've accepted a transaction which contains a key-value pair matching the above conditions in the past, Sentry will suggest values for each condition. Of course, you're free to set a custom value for each condition. Then, specify a sampling rate for that specific rule.

## 4. Increase Your SDK Transaction Sample Rate

Generally speaking, the more active sampling rules you have, the fewer transactions we'll accept. If the volume of transactions our servers are receiving is insufficient given the rules you've specified, you might need to increase the sampling rate configured for your Sentry SDK. To see what we would suggest for your SDK sample rate, go to **[Project] > Settings > Server-side Sampling** and open the last rule in the list, which is where your uniform sample rate is set. Our suggestion is displayed there.

> ## Sampling Migration Guide
> 
> Migrating from Sentry’s old sampling model to the new one? Here’s what’s changing and the things you need to know.
> 
> ### What's Changing
> 
> TBA
> 
> ### What You Need to Do
> 
> TBA

Author: ```suggestion
```

- [Arbitrary logical composition for rule conditions](/product/data-management-settings/server-side-sampling/current-limitations/#arbitrary-logical-composition-for-rule-conditions)

For more detailed information about each of these, check out our [full Current Limitations documentation](/product/data-management-settings/server-side-sampling/current-limitations/).

## 1. Update Sentry SDK Versions

Once you’ve confirmed that your application only relies on [supported SDKs](#current-limitations), then the next step is to update them to the specified version or higher. Check out the migration documentation for the [SDKs](/platforms/) you’re using to learn everything you need to know to get up and running again with the latest Sentry features.

## 2. Set a Uniform Sampling Rate

Next, go to **[Project] > Settings > Server-side Sampling** in [sentry.io](https://sentry.io) and set a uniform sampling rate for all transactions: this is the number of transactions that Sentry should accept from your application:

![Setting a uniform sampling rate for your project](set-uniform-sampling-rate.png)

> Based on your quota and usage, in this step, we’ll suggest sampling rates for both Sentry servers and your Sentry SDK. However, you can enter a custom sampling rate by overriding our recommended values. For example, you can select your current sampling rates to understand how our proposed sampling configuration would impact your transaction volume within the scope of this project. Learn more about [How Sentry Samples](/product/data-management-settings/server-side-sampling/#how-sentry-samples).

Reviewer: > Based on your quota and usage, in this step, we’ll suggest sampling rates for both Sentry servers and your Sentry SDK

"Sentry SDK" sounds strange to me... does it mean only the SDKs supported by us?

Reviewer 2: your in this case refers to the Sentry SDK your project uses to set the sampling rate.

The numbers you enter in the “Client” field do not affect your actual SDK sample rate, but if you click “Next” and save, the number you’ve entered in the “Server” field on the “Suggested” row is saved as your uniform server-side sampling rate.

Even though you’ll only set this for one project/SDK, **this rate can affect all the transactions distributed across multiple services**, including other projects set up in Sentry. That’s because server-side currently samples by the context of all other transactions belonging to the same trace. Check out our [full Tracing documentation](/product/sentry-basics/tracing/) for a more detailed explanation.

If this chart doesn’t provide enough information for you to determine a uniform sampling rate for this project, we recommend checking your **Stats** page. The **Stats** page provides a complete overview of how many transactions each project accepts, filters, and drops rather than for just one project.

> Once you've entered a uniform sample rate for all transactions, now you need to activate it by clicking “Done” and then toggling the rule to on:

Reviewer: ```suggestion
Once you've entered a uniform sample rate for all transactions, now you need to persist it by clicking “Done” and then toggling the rule to on to activate the rule:
```

Reviewer: > now you need to activate it by clicking “Done”

When I first read this... I thought ah ok once I click then it will activate the rule... and only after I read the second sentence, having the AKA moment

Reviewer 2: I see what you're trying to suggest here, but I find the sentence reads better if you introduce the activation part earlier on in the sentence.

---
title: Getting Started
sidebar_order: 10
description: "Learn how to get started with server-side sampling."
---

<Note>

This feature is in beta and is only available if your organization is participating in its limited release. Features in beta are still in-progress and may have bugs. We recognize the irony. If you’re interested in participating, [join the waitlist](https://sentry.io/for/performance/#updates-signup).

</Note>

<Note>

To set up or configure server-side sampling, you need to be an Owner, Manager, or Admin of your Sentry organization.

</Note>

## Current Limitations

Server-side sampling is currently only available for the following SDK's:

- Python: 1.4.0 or later
- JavaScript: 6.13.0 or later
- Apple: 7.23.0 or later

> - Android: 6.2.1 or later

Reviewer 3: Are these versions correct?
For example, the `source` was introduced in JavaScript [7.6.0](https://github.com/getsentry/sentry-javascript/releases/tag/7.6.0) and we need that.

> - Android: 6.2.1 or later

Reviewer 3: Are these versions correct?
For example, the source was introduced in JavaScript [7.6.0](https://github.com/getsentry/sentry-javascript/releases/tag/7.6.0) and we need that.

Reviewer 3: @brustolin do you know what's the correct version for Apple and Android?

Reviewer 4: Yes, thats is correct for Apple and Android

Reviewer 3: ```suggestion
- Python: 1.7.2 or later
- JavaScript: 7.6.0 or later
- Apple: 7.23.0 or later
- Android: 6.2.1 or later
```

---
title: Getting Started
sidebar_order: 10
description: "Learn how to get started with server-side sampling."
---

<Note>

This feature is in beta and is only available if your organization is participating in its limited release. Features in beta are still in-progress and may have bugs. We recognize the irony. If you’re interested in participating, [join the waitlist](https://sentry.io/for/performance/#updates-signup).

</Note>

<Note>

To set up or configure server-side sampling, you need to be an Owner, Manager, or Admin of your Sentry organization.

</Note>

## Current Limitations

Server-side sampling is currently only available for the following SDK's:

- Python: 1.4.0 or later
- JavaScript: 6.13.0 or later
- Apple Cocoa: 7.3.0 or later

> - Java and Android: 5.2.0 or later

Author: ```suggestion
- Java and Android: 6.2.1 or later
```

> <!-- TO DO version numbers need to be confirmed/updated -->

Author: ```suggestion
<!-- TO DO Apple version number needs to be confirmed/updated -->
```

> - Apple Cocoa: 7.3.0 or later
> - Java and Android: 5.2.0 or later
> 
> <!-- TO DO version numbers need to be confirmed/updated -->

Author: ```suggestion
- Apple: 7.23.0 or later
- Android: 6.2.1 or later
```

---
title: Getting Started
sidebar_order: 10
description: "Learn how to get started with server-side sampling."
---

> 

Author: ```suggestion

<Note>

This feature is in beta and is only available if your organization is participating in its limited release. Features in beta are still in-progress and may have bugs. We recognize the irony. If you’re interested in joining this limited release, please reach out at [abc@sentry.io](mailto:abc@sentry.io).

</Note>

```

Once you've entered a uniform sample rate for all transactions, now you need to activate it by clicking “Done” and then toggling the rule to on:

![A sampling rule with the Active toggle highlighted](sampling-rule-toggle.png)

## 3. Set a Sampling Rate Based on a Condition

Now, click “Add Rule” to set more sampling rules based on specific conditions. These rules will help you target the transactions you want to keep and allow you to have more of those transactions sent by the SDK.

When a rule uses multiple conditions, we use the logical expression `AND` to determine when that rule should be evaluated. Multiple values for a condition use a logical or condition for matching. Learn more about [how to create effective server-side sampling rules](/product/data-management-settings/server-side-sampling/#server-side-sampling-configurations).

You can use the following conditions to sample transactions by:

- Environment
- Release\*
- User Segment

> * Allows glob pattern matching

Reviewer 5: i think this will render as bulletpoint, no?

> * Allows glob pattern matching

Author: ```suggestion
\* Allows glob pattern matching
```

> ![Editing a sampling rule based on a release condition](edit-rule-modal.png)

Reviewer 5: the screenshots has a black border (from the drop shadow) when it should be transparent there afaict

Author: > the screenshots has a black border (from the drop shadow) when it should be transparent there afaict

yeah, these are all just placeholders for now and will all be replaced before this is merged.
